design/user_types.go

```diff
@@ -38,6 +38,71 @@ var UpdateWorkItemPayload = a.Type("UpdateWorkItemPayload", func() {
 	a.Required("type", "fields", "version")
 })
 
+// UpdateWorkItemJSONAPIPayload defines top level structure from jsonapi specs
+// visit : http://jsonapi.org/format/#document-top-level
+var UpdateWorkItemJSONAPIPayload = a.Type("UpdateWorkItemJSONAPIPayload", func() {
+	a.Attribute("data", WorkItemDataForUpdate)
+	a.Required("data")
+})
+
+// WorkItemDataForUpdate defines how an update payload will look like
+var WorkItemDataForUpdate = a.Type("WorkItemDataForUpdate", func() {
+	a.Attribute("type", d.String, func() {
+		a.Enum("workitems")
+	})
+	a.Attribute("id", d.String, "ID of the work item which is being updated", func() {
+		a.Example("42")
+	})
+	a.Attribute("attributes", a.HashOf(d.String, d.Any), func() {
+		a.Example(map[string]interface{}{"version": "1", "system.state": "new", "system.title": "Example story"})
+	})
+	a.Attribute("relationships", WorkItemRelationships)
+	// relationships must be required becasue we MUST have workItemType during PATCh
+	a.Required("type", "id", "attributes")
+})
+
+// WorkItemRelationships defines only `assignee` as of now. To be updated
+var WorkItemRelationships = a.Type("WorkItemRelationships", func() {
```

Comment:
thanks, done in https://github.com/almighty/almighty-core/pull/482/commits/22a20be904b51c078ea400b03c95b3729230b003